### PR TITLE
Add Direct Authorization method

### PR DESCRIPTION
**Pull request by proninyaroslav**
https://github.com/prawn-cake/vk-requests/issues/37

### CODE REVIEW

File: vk_requests/session.py [hunk 1]
```diff
@@ -46,6 +48,9 @@ def __init__(self, app_id=None, user_login=None, user_password=None,
         self.interactive = interactive
         self._access_token = None
         self._api_version = api_version
+        self._client_secret = client_secret
+        self.two_fa_supported = two_fa_supported
```
Comment (prawn-cake) [inline vk_requests/session.py]: Any reason why we're not keeping it "private"?
Like this
```suggestion
        self._two_fa_supported = two_fa_supported
```

File: docs/index.md [hunk 2]
```diff
@@ -57,7 +57,19 @@ For example when you got a token on the client side ([implicit flow](https://vk.
 
 Use `service_token` parameter as in the example above. 
 
-   
+### User token with client_secrete (Direct Authorization)
```
Comment (prawn-cake) [inline docs/index.md]: Typo here
```suggestion
### User token with client_secret (Direct Authorization)
```

File: vk_requests/session.py [hunk 3]
```diff
@@ -170,6 +175,78 @@ def do_oauth2_authorization(self, session):
         logger.error('Permissions obtained')
         raise VkAuthError(error_message)
 
+    def do_direct_authorization(self, session):
+        """ Direct Authorization, more info: https://vk.com/dev/auth_direct """
+        logger.info('Doing direct authorization, app_id=%s', self.app_id)
```
Comment (prawn-cake) [inline vk_requests/session.py]: Also "Direct authorization" sounds a bit too general and may be confusing (at least to me). 
Let's mention "Client Credentials Flow" in the message as well

Comment (proninyaroslav) [reply vk_requests/session.py]: `Client Credentials Flow` was used to access secure methods and was deprecated. `Direct Authorization` is completely different.

Comment (prawn-cake) [reply vk_requests/session.py]: Oh, alright. Then we can keep it as it is. Link to the documentation in the method description is helpful enough

**Comment by proninyaroslav** [comment]
> Now `VKSession` gets bigger, I was thinking of some fresh ideas to split its' logic into more composable chunks, for instance use auth methods composition, instead of adding it to the same class.

Yes, the principles of authorization are similar and there is a feeling of code duplication. You can divide it into 3 separate authorizations, but `Service key` looks too small for that.

File: vk_requests/session.py (hunk 1 above)
Comment (prawn-cake) [reply vk_requests/session.py]: I think this change was not applied

File: docs/index.md (hunk 2 above)
Comment (prawn-cake) [reply docs/index.md]: Same here. I guess, you need to press "Commit suggestion" to apply it

File: vk_requests/session.py [hunk 4]
```diff
@@ -46,6 +48,9 @@ def __init__(self, app_id=None, user_login=None, user_password=None,
         self.interactive = interactive
         self._access_token = None
         self._api_version = api_version
+        self._client_secret = client_secret
+        self._two_fa_supported = two_fa_supported
+        self.two_fa_force_sms = two_fa_force_sms
```
Comment (prawn-cake) [inline vk_requests/session.py]: Sorry for not suggesting it in the same comment, but to keep it consistent let's make `two_fa_force_sms` also "private"
```suggestion
        self._two_fa_force_sms = two_fa_force_sms
```

**Comment by prawn-cake** [comment]
I'll merge it and will test locally

**Pull request merged**